specs/accordion.md

```diff
@@ -0,0 +1,193 @@
+# Accordion
+
+## Overview
+
+As defined by the W3C:
+> An accordion is a vertically stacked set of interactive headings that each contain a title, content snippet, or thumbnail representing a section of content. The headings function as controls that enable users to reveal or hide their associated sections of content. Accordions are commonly used to reduce the need to scroll when presenting multiple sections of content on a single page.
+
+### Use Cases
+
+- Frank finds a spelling error in the FAST-DNA documentation. He submits a pull request (PR) to fix the error and goes to eat lunch while waiting for the maintainers to review his PR. Upon returning from lunch he notices a widget on the screen saying certain status checks failed. He clicks on the headline of the widget and it reveals additional information stating that he needs to run prettier to ensure proper formatting. He clicks the headline again and the additional information collapses. Frank runs prettier and his PR is completed successfully.
+
+- On April 16th Randy wakes up in a cold sweat because he realized he made a mistake in filing his taxes with the IRS. He opens his laptop and goes to the IRS website and sees a link to "frequently asked questions". Randy sees a list of questions on the page with an arrow next to it indicating that answers are hidden. He sees a question titled, "What should I do if I made a mistake on my federal turn that I've already filed?" Randy clicks the question and sees a section detailing out the exact steps to take. With his anxiety eased, Randy goes back to bed and falls asleep.
+
+### Features
+
+1. Optionally allow only a single section to be expanded. This behavior is opt-in and not default behavior.
+
+2. Support for interactive content within the header, such as a menu button, checkbox, etc.
+
+3. Optional support for nested accordions. Ideally this behavior will "just work" and no special behaviors will need to be added to enable/support this. The expectation here should be that an accordion takes content and whatever content is inside the accordion panel respects its own interaction model.
+
+### Prior Art/Examples
+
+- [Ant Design](https://ant.design/components/collapse/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/accordion/usage/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/accordion/)
+- [Fluent UI](https://fluentsite.z22.web.core.windows.net/components/accordion/accessibility)
+- [Primer](https://primer.style/components/Details)
+- [Bootstrap](https://getbootstrap.com/docs/4.3/components/collapse/)
+
+---
+
+### API
+
+**Accordion**
+*Component Name*
+- `fast-accordion`
+
+*Attributes:*
```

Comment:
It might be good to land a convention regarding how we name attributes with multiple words (`"expandmode"` vs `"expand-mode"`, `"headinglevel"` vs `"heading-level"`) so that we don't deviate around the project. For instance, the DesignSystemProvider uses dashes to delineate word-breaks: `"background-color"`.

@microsoft/fast-dna-collaborators thoughts and preferences?

Reply:
Personally I think to be more inline with HTML attributes we **should not** use the hyphens.

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes

Interesting there is only one multiword attribute in the about list that uses a hyphen `accept-charset`.

Reply:
For `dataset` and data attributes, dashes (after the initial `data-`) suggest arrays or lists of properties.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/dataset

Reply:
> ...suggest arrays or lists of properties

I'm not follow this, can you clarify? Is there an example somewhere?

My takeaway from the above dataset docs is that a data *attribute name* (eg `data-foo-bar`) would be accessible via the dataset property by the *property name* `fooBar` whereas `data-foobar` would be accessible on the property `foobar`. I think it would follow from that then that a camel-cased property would reflect as a hyphen separated attribute name. While I prefer that convention, there are native properties in HTML that *don't* follow that pattern (`readonly` vs `readOnly`) 

I personally would prefer to see hyphen separated simply because it is easier to read, especially for longer attribute names that concatenate 3+ words (eg `"basehorizontalspacingmultiplier"` vs `"base-horizontal-spacing-multiplier"`)

Reply:
> I'm not follow this, can you clarify? Is there an example somewhere?

My bad, that part was from a different thing entirely. I confused the functionality of a custom implementation with the expected behavior for kebab-to-camelCase conversion.

Reply:
I'm ok with adding hyphens, Is this something we should talk about retroactively fixing in other components?

Reply:
@eljefe223 Probably, though I was hoping others would chime in. I'll open an issue to discuss and we can address components as needed based on where that shakes out. Thanks!